File: frontend/src/dtos/SalesDtos.ts
// DTOs relacionados con el módulo de ventas

// DTO para los detalles de una venta (linea de venta)
export interface SaleDetailDto {
  saleDetailId?: string;
  productId: string;
  quantity: number;
  unitPrice: number;
}

// DTO para crear un detalle de venta
export interface CreateSaleDetailDto {
  productId: string;
  quantity: number;
  unitPrice: number;
}

// DTO para crear una venta completa
export interface CreateFullSaleDto {
  branchId: string;
  totalAmount: number;
  saleDetails: CreateSaleDetailDto[];
}

// DTO para la información de una venta
export interface SaleDto {
  saleId: string;
  branchId: string;
  userId: string;
  totalAmount: number;
  saleDate: string;
  createdDate?: string;
  updatedDate?: string;
  saleDetails: SaleDetailDto[];
}

// Interface para el estado de la página de ventas
export interface SalesPageState {
  loading: boolean;
  createModalVisible: boolean;
  detailModalVisible: boolean;
  selectedSale: SaleDto | null;
  products: ProductDto[];
  sales: SaleDto[];
  search: string;
  pagination: {
    current: number;
    pageSize: number;
    total: number;
  };
}

// DTO simplificado para productos (usado en el módulo de ventas)
export interface ProductDto {
  ProductId: string;
  ProductName: string;
  ProductPrice: number;
  ProductDescription?: string;
  ProductBarcode?: string;
} 

File: backend/Dtos/SalesDtos.cs
using System.ComponentModel.DataAnnotations;
using CRM_ERP_UNID.Attributes;

namespace CRM_ERP_UNID.Dtos;

public class CreateFullSaleDto
{
    [Required]
    [GuidNotEmpty]
    public Guid branchId { get; set; }

    [Required]
    [Range(0.01, double.MaxValue)]
    public decimal totalAmount { get; set; }
    
    [MinLength(1)]
    [Required]
    public List<CreateSaleDetailDto> SaleDetails { get; set; } = new();
}

public class SaleDto
{
    [Required]
    [GuidNotEmpty]
    [IsObjectKey]
    public Guid id { get; set; }

    [Required]
    [GuidNotEmpty]
    public Guid branchId { get; set; }

    [Required]
    [GuidNotEmpty]
    public Guid userId { get; set; }

    [Required]
    public decimal totalAmount { get; set; }

    [Required]
    public DateTime? saleDate { get; set; }

    public DateTime? createdDate { get; set; }
    public DateTime? updatedDate { get; set; }
}